### ISSUE TITLE
Save and Load model in R not saving num_class parameter

### ISSUE THREAD

**Issue by titericz** [dmlc/xgboost#232]
I'm using R. I trainned one model and then saved it to disk using xgb.save(bst, filename).
Then I load it using bst <- xgb.load( filename ) with no erros.
 But when I try to predict using that loaded model it rises an error saying that num_class must be set first. 
Probably that parameter is not being saved in the model file.
 If I predict before saving to file it predicts ok.  My model is multiclass. 
Is it an issue? Is there any other way to set that parameter after load a model from file? Thks!

**Comment by tqchen** [comment]
It would be great if you can provide a short snippet of code to reproduce the error, it does not have to be real data

**Comment by titericz** [comment]
Here it is! Could you take a look?


rm(list = ls(all = TRUE))

#Create random dataset and multiclass target
train <- matrix( runif(10000,-1,1) , nrow=100,ncol=100  )
target<- ifelse( rowSums(train)>0, ifelse( rowSums(train)>2, 2 , 1 ) , 0 )
table(target)

xgmat   <- xgb.DMatrix( train[1:50,], label = target[1:50])
param <- list("objective" = "multi:softprob",
              "num_class" = 3 ,
              "bst:eta" = 0.01,
              "bst:max_depth" = 3 ,
              "subsample" = 1.0,
              "colsample_bytree" = 1.0 ,
              "silent" = 1,
              "nthread" = 4)
bst = xgb.train(param, xgmat, 100)
xgmatVAL   <- xgb.DMatrix( train[51:100,])
y <- predict(bst, xgmatVAL)
xgb.save( bst, 'model.xgb' )


bst <- xgb.load( 'model.xgb' )
y <- predict(bst, xgmatVAL)    <<<=====   Error in .local(object, ...) : must set num_class to use softmax

If I change "objective" to "binary:logistic" and change the target to be binary it works.
 Regards,Gilberto

      De: Tianqi Chen <notifications@github.com>
 Para: dmlc/xgboost <xgboost@noreply.github.com> 
Cc: Gilberto Titericz Junior <titericz@yahoo.com> 
 Enviadas: Quinta-feira, 9 de Abril de 2015 12:49
 Assunto: Re: [xgboost] Save and Load model in R not saving num_class parameter (#232)
   
It would be great if you can provide a short snippet of code to reproduce the error, it does not have to be real data—

**Comment by hetong007** [comment]
Hi @titericz , I just installed the latest version of xgboost on my CentOS machine and tried your code. There's no warning. Do you mind to check it again? Thank you.

**Comment by titericz** [comment]
I tested and now its working fine. Thanks! 

<div>-------- Mensagem original --------</div><div>De : Tong He <notifications@github.com> </div><div>Data:27/04/2015  20:24  (GMT-03:00) </div><div>Para: dmlc/xgboost <xgboost@noreply.github.com> </div><div>Cc: Gilberto Titericz Junior <titericz@yahoo.com> </div><div>Assunto: Re: [xgboost] Save and Load model in R not saving num_class parameter
  (#232) </div><div>
</div>Hi @titericz , I just installed the latest version of xgboost on my CentOS machine and tried your code. There's no warning. Do you mind to check it again? Thank you.
